Sources/KituraNet/HTTP/HTTPServerResponse.swift

```diff
@@ -115,13 +118,14 @@ public class HTTPServerResponse: ServerResponse {
     ///
     public func end() throws {
         guard let channel = self.channel else {
-            fatalError("No channel handler context available.")
+            fatalError("No channel available.")
         }
-        if channel.eventLoop.inEventLoop {
-            try end0(channel: channel)
-        } else {
-            channel.eventLoop.execute {
-                try! self.end0(channel: channel)
+
+        execute(on: channel.eventLoop) {
+            do {
+                try self.end0(channel: channel)
+            } catch let error {
+                fatalError("Error: \(error)")
```

Comment:
@pushkarnk  should we crash here?

Reply:
I don't think we should crash at these places. We must simply `Log.error()` and return. We need to do all of that in a separate PR though.